### PR TITLE
add mavenFeatures and repository.generator projects

### PR DESCRIPTION
**Pull request by rampras3**

### CODE REVIEW

**Comment by rampras3** [comment]
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_pd8qECKJEeiwitX0Zm6mOA 

Errors in build are unrelated to changes.

**Comment by rampras3** [comment]
Personal Build: 
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_Um-7QCLWEeiwitX0Zm6mOA

File: dev/com.ibm.ws.repository.generator/build.gradle [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                fileset(dir: "${rootDir}/com.ibm.ws.repository.parsers/build/libs") {
+                    include(name: 'com.ibm.ws.repository.parsers.jar')
+                }
+                fileset(dir: "${rootDir}/com.ibm.ws.logging.core/build/libs") {
+                    include(name: 'com.ibm.ws.logging.core.jar')
+                }
+                fileset(dir: "${rootDir}/com.ibm.ws.kernel.feature.core/build/libs") {
+                    include(name: 'com.ibm.ws.kernel.feature.core.jar')
+                }
+             //   fileset(dir: "${rootDir}/*/build/libs/*") {
```
Comment (ericglau) [inline dev/com.ibm.ws.repository.generator/build.gradle]: Should this be removed?

File: dev/com.ibm.ws.repository.generator/src/com/ibm/ws/repository/generator/SingleJsonRepositoryGenerator.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        }
+        
+        SingleFileRepositoryConnection mySingleFileRepo = null;
+        if (jsonFile.exists()) {
+        	mySingleFileRepo = new SingleFileRepositoryConnection(jsonFile);
+        } else {
+        	try {
+				mySingleFileRepo = SingleFileRepositoryConnection.createEmptyRepository(jsonFile);
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
```
Comment (ericglau) [inline dev/com.ibm.ws.repository.generator/src/com/ibm/ws/repository/generator/SingleJsonRepositoryGenerator.java]: Does this exception need to be handled?

File: dev/com.ibm.ws.repository.generator/src/com/ibm/ws/repository/generator/SingleJsonRepositoryGenerator.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+                //resource.writeDiskRepoJSONToStream(os);
+            } else {
+                // Not a file we build json for
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new BuildException(e);
+        }
+        
+//        finally {
```
Comment (ericglau) [inline dev/com.ibm.ws.repository.generator/src/com/ibm/ws/repository/generator/SingleJsonRepositoryGenerator.java]: Should this be removed?  (As well as the other commented out sections in this file)

File: dev/com.ibm.ws.repository.generator/build.gradle [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    dependsOn parent.subprojects.assemble
+    from cnf.fileTree(".").include("/release/dev/*/*/*.esa").files
+    into 'build/esas'
+}
+
+task createSingleJson {
+    // Run after all feature ESAs have been built
+    dependsOn copyEsas
+    doLast {
+        ant.taskdef(name:'singleJsonRepo', classname:'generator.SingleJsonRepositoryGenerator') {
```
Comment (ericglau) [inline dev/com.ibm.ws.repository.generator/build.gradle]: Does this work?  Looks like classname should be com.ibm.ws.repository.generator.SingleJsonRepositoryGenerator

File: dev/com.ibm.ws.wlp.mavenFeatures/build-buildtools.xml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project name="LibertyFeaturesToMavenRepo" default="publish">
```
Comment (ericglau) [inline dev/com.ibm.ws.wlp.mavenFeatures/build-buildtools.xml]: Change project name to com.ibm.ws.wlp.mavenFeatures

File: dev/com.ibm.ws.wlp.mavenFeatures/default.properties [hunk 6]
```diff
@@ -0,0 +1 @@
+libertyFeaturesToMavenRepo=generator.LibertyFeaturesToMavenRepo
```
Comment (ericglau) [inline dev/com.ibm.ws.wlp.mavenFeatures/default.properties]: The value should be com.ibm.ws.wlp.mavenFeatures.LibertyFeaturesToMavenRepo ?

File: dev/com.ibm.ws.repository.generator/src/com/ibm/ws/repository/generator/SingleJsonRepositoryGenerator.java (hunk 2 above)
Comment (rampras3) [reply dev/com.ibm.ws.repository.generator/src/com/ibm/ws/repository/generator/SingleJsonRepositoryGenerator.java]: Exception is handled in createEmptyRepository.

**Comment by ericglau** [comment]
This is being handled by other pull requests.

**Pull request closed without merging**